DiscordChatExporter.Core/Exporting/Filtering/MessageFilter.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using DiscordChatExporter.Core.Discord.Data;
+using DiscordChatExporter.Core.Exporting.Filtering.Parsing;
+using Superpower;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public abstract partial class MessageFilter
+    {
+        public abstract bool Filter(Message message);
+    }
+
+    public partial class MessageFilter
+    {
+        public static MessageFilter CreateFilter(string text) => new ContainsMessageFilter(text);
+
+        public static MessageFilter CreateFilter(string key, string value)
```

Comment:
These two probably don't need to be exposed?

```suggestion
        private static MessageFilter CreateFilter(string text) => new ContainsMessageFilter(text);

        private static MessageFilter CreateFilter(string key, string value)
```

Reply:
Yeah let's make them internal